packages/newspringchurchapp/src/prayer/PrayerList/index.js

```diff
@@ -81,158 +76,115 @@ class PrayerList extends PureComponent {
   };
 
   // TODO: this may be better pulled from a content channel???
-  calculateQuery = () => {
-    let queryTag;
-    let label;
-    let variables;
-
-    switch (this.props.query) {
-      case 'groups':
-        queryTag = GET_GROUP_PRAYERS;
-        label = 'My Community';
-        break;
-      case 'campus': {
-        const {
-          currentUser: { profile: { campus: { id } = {} } = {} } = {},
-        } = cache.readQuery({
-          query: getUserProfile,
-        });
-        queryTag = GET_CAMPUS_PRAYERS;
-        label = 'My Campus';
-        variables = { campusId: id };
-        break;
-      }
-      default:
-        queryTag = GET_PRAYERS;
-        label = 'My Church';
-        break;
-    }
-    return { queryTag, label, variables };
-  };
 
   render() {
-    const { queryTag, label, variables } = this.calculateQuery();
+    const { navigation } = this.props;
```

Comment:
```suggestion
    const { title, prayers, query } = this.props.navigation.state.params;
```

Reply:
and this could probably go outside the `render()` method

Reply:
I like your thought here, but I think we still need `navigation` further down (for the `pop` on the modal and it looks like it's getting passed to the `PrayerSingle`).

Reply:
Use `this.props.navigation` takes one less line :)